PLBenchmarks/ligands.py

```diff
@@ -0,0 +1,327 @@
+"""
+ligands.py
+Functions and classes for handling the ligand data.
+"""
+
+import os
+import re
+import io
+import yaml
+import pandas as pd
+
+from PIL import Image
+from rdkit import Chem
+from rdkit.Chem import Draw, PandasTools
+from openforcefield.topology import Molecule
+
+import PLBenchmarks
+import PLBenchmarks.targets
+import PLBenchmarks.utils
+
+
+class ligand:
+    """
+    Store and convert the data of one ligand in a :py:class:`pandas.Series`.
+
+    """
+
+    _observables = ["dg", "dh", "tds", "ki", "ic50", "pic50"]
+
+    def __init__(self, d: dict, target: str = None):
+        """
+        Initialize :py:class:`PLBenchmarks.ligands.ligand` object from :py:class:`dict` and store data in a :py:class:`pandas.Series`.
+
+        :param d: :py:class:`dict` with the ligand data
+        :return None
+        """
+        self._target = target
+        self._data = pd.Series(d)
+        self._molecule = None
+        self._name = self._data["name"]
+        # Expand measurement dict into single fields
+        if "measurement" in list(self._data.index):
+            meas = pd.Series(self._data["measurement"])
+            meas.index = ["measurement:" + c for c in meas.index]
+            self._data.drop(["measurement"], inplace=True)
+            self._data = pd.concat([self._data, meas])
+            index = self._data.index.to_series().str.split(":", expand=True).fillna("")
+            self._data.index = pd.MultiIndex.from_arrays(
+                [index[0].tolist(), index[1].tolist()]
+            )
+            for obs in self._observables:
+                if ("measurement", obs) in list(self._data.index):
+                    self._data = self._data.append(
+                        pd.Series(
+                            [0],
+                            index=pd.MultiIndex.from_tuples(
+                                [("measurement", f"e_{obs}")]
+                            ),
+                        )
+                    )
+                    vals = self._data[("measurement", f"{obs}")]
+                    if vals[2] == "nM":
+                        u = PLBenchmarks.utils.ureg("nanomolar")
+                    elif vals[2] == "uM":
+                        u = PLBenchmarks.utils.ureg("micromolar")
+                    elif vals[2] == "kj/mol":
+                        u = PLBenchmarks.utils.ureg("kJ / mole")
+                    elif vals[2] == "kcal/mol":
+                        u = PLBenchmarks.utils.ureg("kcal / mole")
```

Comment:
For all of these but `kj/mol` (because of `kj` vs `kJ`), a default `pint.UnitRegistry` parses them  as expected. Could this be condensed down a little bit?

Reply:
Condensed in commit 6931106.